plugins/modules/clustering/consul/consul.py

```diff
@@ -475,6 +494,18 @@ def __init__(self, check_id, name, node=None, host='localhost',
 
             self.check = consul.Check.http(http, self.interval, self.timeout)
 
+        if tcp:
+            if interval is None:
+                raise Exception('tcp check must specify interval')
+
+            regex = r"(?P<host>.*)(?::)(?P<port>(?:[0-9]+))$"
+            match = re.match(regex, tcp)
+
+            if match is None:
+                raise Exception('tcp check must be in host:port format')
+
+            self.check = consul.Check.tcp(match.group('host'), int(match.group('port')), self.interval)
```

Comment:
```suggestion
            self.check = consul.Check.tcp(match.group('host').strip('[]'), int(match.group('port')), self.interval)
```
In case the IPv6 syntax `[1:2::3]:80` is used, which is rather common for combining IPv6 addresses with ports.